src/game_api/strings.cpp

```diff
@@ -151,6 +151,17 @@ const char16_t* get_string(STRINGID string_id)
     return strings_table[string_id];
 }
 
+STRINGID pointer_to_stringid(size_t ptr)
+{
+    auto strings_table = get_strings_table();
+    for (STRINGID i = 0; i < g_original_string_ids_end; ++i)
+    {
+        if ((size_t)strings_table[i] == ptr)
+            return i;
+    }
+    return g_original_string_ids_end;
```

Comment:
This can be solved with simple math, just subtract the ptr from address of the stings_table and divide that by 8 (or something like that)

Reply:
But the table is ** and what I have is the last *

Reply:
oh wait, i dumb

Reply:
It's not actually always even in the strings table, when the string needed formatting for example. But it works for this very specific need of converting death messages to stringids...